apps/src/tutorialExplorer/responsive.jsx

```diff
@@ -0,0 +1,29 @@
+function getContainerWidth(windowWidth) {
+  if (windowWidth >= 1200) {
+    return 1170;
+  } else if (windowWidth >= 992) {
+    return 970;
+  } else if (windowWidth >= 768) {
+    return 750;
+  } else {
+    return "100%";
+  }
+}
+
+function getItemWidth(itemWidth, windowWidth) {
+  if (windowWidth >= 600) {
+    return `${itemWidth}%`;
+  } else {
+    return "100%";
+  }
+}
+
+function getItemWidthMobile(itemWidth, windowWidth) {
+  if (windowWidth < 600) {
+    return `${itemWidth}%`;
+  } else {
+    return "100%";
+  }
+}
+
+export { getContainerWidth, getItemWidth, getItemWidthMobile };
```

Comment:
Rather than passing around windowWidth as a prop, I might just have your responsive methods here calculate it (Means you don't have to worry about getting a mismatch between what a component thinks the window widht is and what these method thinks it is).

Downside is that it makes these methods less pure, as they now depend on browser, but that seems okay. I'd also add comments above these explaining (a) where these magic numbers come from and (b) what you're trying to accomplish.